pytest/tests/mocknet/outage.py

```diff
@@ -0,0 +1,107 @@
+# Stop fewer than one third of mocknet nodes for two full epochs.
+# During this time the network should continue to function.
+# The stopped nodes are then restarted, should come back up,
+# sync successfully and become operational again.
+
+import sys, time
+from rc import pmap
+
+sys.path.append('lib')
+import mocknet
+
+nodes = mocknet.get_nodes()
+
+validators = nodes[0].get_validators()['result']
+validator_accounts = set(
+    map(lambda v: v['account_id'], validators['current_validators']))
+# Nodes 0, 1, 2, 3 are initially validators
+assert validator_accounts == {'node0', 'node1', 'node2', 'node3'}
+
+epoch_length = mocknet.get_epoch_length_in_blocks(nodes[0])
+
+# Function to do a transfer, but not using the first node
+check_transfer = lambda: mocknet.transfer_between_nodes(nodes[1:])
+
+
+# Function to wait until a particular block height (used to wait for next epoch to start)
+def wait_until_height(target_height, query_node):
```

Comment:
it seems that you rather want to wait until the epoch_start_height changes. An epoch can be longer in heights if for example blocks are not getting finalized.

Reply:
Done in https://github.com/nearprotocol/nearcore/pull/2868/commits/ab35f41470c40ebf8881caf85f9cd4ae477b52de